src/function_data.jl

```diff
@@ -285,12 +287,11 @@ Base.length(pwl::Union{PiecewiseLinearData, PiecewiseStepData}) =
 Base.getindex(pwl::PiecewiseLinearData, ix::Int) =
     getindex(get_points(pwl), ix)
 
-Base.:(==)(a::PiecewiseLinearData, b::PiecewiseLinearData) =
-    get_points(a) == get_points(b)
+Base.:(==)(a::T, b::T) where {T <: FunctionData} = double_equals_from_fields(a, b)
```

Comment:
The implementation is good. Are there any differences with https://github.com/JuliaServices/AutoHashEquals.jl? If they are equivalent, any reason not to use that package? I'm not opposed to having this here.

You might want to mention Julia issue 4648. Seems like the default behavior may change in the future.

Reply:
I think AutoHashEquals should work — @jd-lara okay to add as a dependency?

Reply:
If don't need it and this works we should keep our own method instead of more dependencies.